src/encode.cpp

```
@@ -957,6 +957,64 @@ void encodeOp(State &st, mlir::tosa::BitwiseXorOp op, bool) {
       [](auto &&a, auto &&b) { return (Expr)a ^ (Expr)b; });
 }
 
+template<>
+void encodeOp(State &st, mlir::tosa::TransposeOp op, bool) {
+  auto dty = op.getType().dyn_cast<mlir::RankedTensorType>();
+  if (!dty)
+    throw UnsupportedException(op.getOperation(), "Unsupported type");
+
+  mlir::Value i = op.input1();
+  mlir::Value p = op.perms();
+
+  auto ity = i.getType().dyn_cast<mlir::RankedTensorType>();
+  auto pty = p.getType().dyn_cast<mlir::RankedTensorType>();
+  if(!getElemTy(p).isa<mlir::IntegerType>())
+    throw UnsupportedException(op.getOperation(), "Unsupported element type");
+
+  assert(pty.getRank() == 1 && pty.getDimSize(0) == ity.getRank());
+
+  auto input = st.regs.get<Tensor>(i);
+  auto perms = st.regs.get<Tensor>(p);
+
+  vector<Expr> indVars = Index::boundIndexVars(input.getRank());
+  vector<Expr> dims, outVars;
+  vector<uint64_t> idxs;
+
+  for (unsigned i = 0; i < input.getRank(); i++) {
+    uint64_t v;
+    // We expect simplify() to succeed since perms is a small Tensor
+    assert(perms.get({Index(i)}).first.simplify().isUInt(v));
```

Should we throw `UnsupportedException` rather than `assert`?